### PR TITLE
command/fmt: Restore some more-opinionated behaviors

### PR DESCRIPTION
**Pull request by apparentlymart**
In Terraform 0.11 and earlier, the `terraform fmt` subcommand was very opinionated in the interests of consistency. While that remains its goal (similar to `go fmt` in the Go community), for pragmatic reasons Terraform 0.12 significantly reduced the number of formatting behaviors in the `fmt` sub command. We've held off on introducing 0.12-and-later-flavored cleanups out of concern it would make it harder to maintain modules that are cross-compatible with both Terraform 0.11 and 0.12, but with this aimed to land in 0.14 -- two major releases after 0.12 -- our new goal is to help those who find older Terraform language examples have them updated to the more modern idiom.

More rules may follow later, now that the implementation is set up to allow modifications to tokens as well as modifications to whitespace, but for this initial pass the command will now apply the following small set of formatting conventions:

- 0.11-style quoted variable type constraints will be replaced with their 0.12 syntax equivalents. For example, `"string"` becomes just `string`. (This change quiets a deprecation warning.)
- Collection type constraints that don't specify an element type will be rewritten to specify the `any` element type explicitly, so e.g. `list` becomes `list(any)`.
- Arguments whose expressions consist of a quoted string template with only a single interpolation sequence inside will be "unwrapped" to be the naked expression instead, which is functionally equivalent. For example, `"${var.foo}"` becomes just `var.foo`. (This change quiets a deprecation warning.)
- Block labels are given in quotes.

Two of the rules above are coming from a secondary motivation of continuing down the deprecation path for two existing warnings, so authors can have two active deprecation warnings quieted automatically by `terraform fmt`, without the need to run any third-party tools.

All four of these rules match with current documented idiom as shown in the Terraform documentation, so anyone who has followed the documented style should see no changes as a result of this. Those who have adopted other local style will see their configuration files rewritten to the standard Terraform style, but it should not make any changes that affect the _meaning_ of the configuration.

There are some further similar rewriting rules that could be added in future, such as removing 0.11-style quotes around various keyword or static reference arguments, but this initial pass focused on some rules that have been proven out in my third-party tool [`terraform-clean-syntax`](https://github.com/apparentlymart/terraform-clean-syntax), from which much of this commit is a direct port.

For now this doesn't attempt to re-introduce any rules about vertical whitespace, even though the 0.11 `terraform fmt` would previously apply such changes. We'll be more cautious about those, because the results of the rules in Terraform 0.11 were often sub-optimal and so we'd prefer to re-introduce those with some care to the implications for those who may be using vertical formatting differences for some semantic purpose, like grouping together related arguments.

### CODE REVIEW

File: command/testdata/fmt/variable_type_in.tf
Comment (mildwonkey) [inline command/testdata/fmt/variable_type_in.tf:1]: What do you think about adding an object type variable in here? I ask mostly for completion, as I'm someone who likes to read tests first when looking at code, and it'd be nice (even though the expected behavior seems pretty obvious) to have all that in here as well. You absolutely don't need to block merging on this request if you don't want to.

File: command/fmt.go
Comment (mildwonkey) [inline command/fmt.go:414]: Thank you for the excellent explanation here; I can already see this saving someone (me!) from making an unintentionally unfortunate change.

**Review by mildwonkey — APPROVED** [review]
this all looks great! I left one teeny request in the test files, but it's completely fine to ignore. As usual your comments and test documentation are great and I really appreciate it (and you).

File: command/testdata/fmt/variable_type_in.tf
Comment (apparentlymart) [reply command/testdata/fmt/variable_type_in.tf:1]: I assume by this you mean some more examples of situations that currently _aren't_ significantly changed by `terraform fmt`, which would then potentially help us as we add additional behaviors in future? If so, that sounds like a great idea and I'll put some in now before I merge this.

**Pull request merged**
